### DISCUSSION TITLE
Support for Transcript

### DISCUSSION THREAD

**Discussion by AlmightyLks** [Tyrrrz/YoutubeExplode#619] [Q&A] [answered]
Simple question:

Can you fetch a video's transcription through YoutubeExplode?
If not, why not?

**Comment by Tyrrrz** [comment, marked as answer]
You mean closed captions?

**Reply by AlmightyLks** [reply]
![image](https://user-images.githubusercontent.com/54688434/153097785-d8fff470-5300-43b0-80b8-ef547d68d644.png)
![image](https://user-images.githubusercontent.com/54688434/153097846-4810483a-16b7-4ce6-b099-effbf413001d.png)

Never thought it through tbh, but the transcript should be the captions and its timestamps, yeah

**Reply by AlmightyLks** [reply]
"Closed Captions" was a good keyword, found my way around :)
Thank you

**Comment by AlmightyLks** [comment]
@Tyrrrz 
Any reason why captions can't be fetched for https://www.youtube.com/watch?v=1VKIIw05JnE ?
"Could not extract caption duration" is the only information given to me by the exception, looking at the captions / transcript on the site manually works

**Reply by Tyrrrz** [reply]
Not sure

**Reply by AlmightyLks** [reply]
You're parsing the p-tag's attributes of a caption here: https://github.com/Tyrrrz/YoutubeExplode/blob/0ae2b0c92835efd657ed724db8f52adb9bb916c6/YoutubeExplode/Bridge/ClosedCaptionExtractor.cs#L24-L26

When it doesn't find it however, it throws, here:
https://github.com/Tyrrrz/YoutubeExplode/blob/46d47b5785d7c203fc167dc3cdb3cf2fccaa055f/YoutubeExplode/Videos/ClosedCaptions/ClosedCaptionClient.cs#L85-L87

Looking at https://js-jrod.medium.com/the-first-complete-guide-to-youtube-captions-f886e06f7d9d#6aa6
It states that the d-attribute is mandatory. However, I come across the case that it is missing
![image](https://user-images.githubusercontent.com/54688434/153252629-a94fb8de-49ed-40e6-a264-90c84af3c1d0.png)

Which is where it ends up breaking for me.
Have you come upon this before and/or what is your opinion on how things are?



Not sure whether you don't care about this feature at all or you'd be interested in a change

**Reply by Tyrrrz** [reply]
Hmm, I could've sworn we skipped captions with no duration 🤔 
Can you please submit a PR to change the code you linked [ClosedCaptionClient.cs](https://github.com/Tyrrrz/YoutubeExplode/blob/46d47b5785d7c203fc167dc3cdb3cf2fccaa055f/YoutubeExplode/Videos/ClosedCaptions/ClosedCaptionClient.cs#L85-L87) to continue instead of throwing? And also please add your video to the test suite for regression purposes.

**Reply by AlmightyLks** [reply]
Mind elaborating on "add your video to the test suite"?
Would you like me to add a new test for "download and skip borked auto-generated captions"?

**Reply by AlmightyLks** [reply]
@Tyrrrz

**Reply by Tyrrrz** [reply]
Similar to https://github.com/Tyrrrz/YoutubeExplode/blob/4ef7d50fbbdca2a4ae7c0d45a227c36a0f4e0e55/YoutubeExplode.Tests/ClosedCaptionSpecs.cs#L49-L63 except something like `User_can_get_a_specific_closed_caption_track_from_a_video_with_broken_autogenerated_captions`
